Yoast/javascript
# LINGO-328 Merge Hebrew feature branch

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds Hebrew papers and specs to check the assessments functionality for Hebrew.
* [yoastseo] Adds the consecutive sentences assessment for Hebrew.
* [yoastseo] Adds Sentence length assessment for Hebrew.
* [yoastseo] Implements the transition words assessment for Hebrew.
* [yoastseo] Adds passive voice assessment for Hebrew, including a check, a number or specs and a list of Hebrew verb roots, both in their original and (for some) modified version.
* [yoastseo] Implements Hebrew stemmer.
* [Yoast SEO Free] Adds the consecutive sentences assessment for Hebrew.
* [Yoast SEO Free] Adds the sentence length assessment for Hebrew.
* [Yoast SEO Free] Adds the assessment for transition words in Hebrew.
* [Yoast SEO Free] Adds the assessment for the recognition of the passive voice in Hebrew.
* [Yoast SEO Premium] Introduces word form support for Hebrew.


## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

### Extra test steps for developers developers:
* Run yarn test and make sure all tests pass.

### Testing instructions for QA
* Set your website to Hebrew (עברית).
* To create Hebrew text you can use e.g., https://generator.lorem-ipsum.info/_hebrew

   * In Yoast Seo Free:

      1. For the sentence beginning assessment: 

         * Create a new post and add three or more Hebrew sentences:
         * that all begin with the same written-out number, like `אחד` ("one" in masc. form)
       E.g. `.אחד פרי מונח על השולחן. אחד פירות על השולחן. אחד פירות על השולחן`
         * that all begin with the same demonstrative pronoun, like `זה`('this').
       E.g. `.זה המסמך מוכן. זה המשחק נגמר. זה הסרט התחיל`
         * In each of the above cases, the sentence beginning readability assessment should read: `Consecutive sentences: There is enough variety in your sentences. That's great!`.
         * Write three or more Hebrew sentences that begin with the same word.
          E.g. `.שלום מה שלומך? שלום אני בסדר. שלום עולם! שלום לכל אחד מכם`
         * The sentence beginning readability assessment should read: `Consecutive sentences: The text contains 4 consecutive 
          sentences starting with the same word. Try to mix things up!`.


      2. For the sentence length assessment:
         * Create a new post and add a text with some Hebrew sentences.
         * Make sure that your sentence length assessment score reads `Sentence length: Great!`. If thats not yet the case, add `.` between sentences until you get a good result
         * Merge some sentences by removing "." between sentences.
         * Check that the Sentence length assessment score becomes worse, e.g. `Sentence length: 27.3% of the sentences contain more than 15 words, which is more than the recommended maximum of 25%. Try to shorten the sentences.`
         * Split some sentence by adding ".".
         * Check that the Sentence length assessment score improves and again read `Sentence length: Great!`.

     3. For the transition words assessment: 

        * Create a new post and add this text containing the transition word `גם`:
      `.העיר מקודשת ביהדות, בנצרות ובאסלאם. גם היא הייתה מרכז חיי העם היהודי בימי קדם, ומושא כמיהתו וכיסופיו בעת גלותו`
        * The assessment bullet should be green and read: `Transition words: Well done!`
        * Out of the two sentences, only the second ( `גם היא הייתה מרכז חיי העם היהודי בימי קדם, ומושא כמיהתו וכיסופיו בעת גלותו`) should be highlighted.

     4. For the passive voice assessment:

        * Create a new post and add these sentences to your post:
      `.מעל גופתו של קדוש מבורך, הוא הומלך למלך`
      `!כאשר שטן גורש מ שמיים, הוא קולל לזחול כדור הארץ כמו נחש`
      `.וכתביו בנושאי פילוסופיה ייקראו במשך דורות`
        * Check that the passive voice bullet turns red.
                                                                                           
* In Yoast SEO Premium:
   * Add a Hebrew text of at least 150 words.
   * Set `איחרנו` as keyphrase and enter `יאחר` in the text/meta description/title.  The relevant assessments should recognise 
    `יאחר ` as an instance of the keyphrase. You should see the following assessment feedback:
      * `Keyphrase in meta description: Keyphrase or synonym appear in the meta description. Well done!`
      * `Keyphrase density: The focus keyphrase was found 1 time. That's less than the recommended minimum of 3 times for a text of this length. Focus on your keyphrase!` (Number depending on how often you added the keyphrase in the text)
      * `Keyphrase in title: Does not contain the exact match. Try to write the exact match of your keyphrase in the SEO title and put it at the beginning of the title.`


   * Switch the keyphrase and the form in the text/meta description/title around and make sure the assessment results are still 
     the same.
   * Repeat the same process with `ישלחו` as keyphrase and `ששלחה` in the text/meta description/title.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-328
